### ISSUE TITLE
Bug: False positive of error CS8347: Cannot use a result of 'Expression' in this context because it may expose variables referenced by parameter 'x' outside of their declaration scope

### ISSUE THREAD

**Issue by bernd5** [dotnet/roslyn#52252]
**Version Used**: 16.10.0 Preview 1.0

**Steps to Reproduce**:

1. Compile the following code:

```CSharp
using System;

static ref string CalcValue(out int x)
{
    x = 42;
    return ref Strings.s;
}

ref var value = ref new ByRefFunc<string>(() =>
{
    int x;
    return ref CalcValue(out x);
})();


internal delegate ref T0 ByRefFunc<T0>();

class Strings{
   public static string s = "Something";
}
```
[Link](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxAgrgOwD4HpcDCA9gCZwB0AxlCALABQAAgEwCMDDjrAbAAQJwAZry4AGXgQgAbSgDVp6OAAoi6GLwCWmdQA8AlAwDeDXqd47eAXl4AWZgG4TZxgHZ+Q3gGUYCLQHMocihHegBfDnoBYQA3CAReWKlFK3dhTDgAd14AIQBPACUhRgBWAB4xAD4lJT0rCqMnUy1dELMRNyiJaTkFZVVdPRDQvRqQiObETGleMik4Pwh4VN4AFXE8wsES0rWqwYiWLx9/KGN6ZwBmER5r8SgUgCJPIjA4GAALfwehoA==)

**Expected Behavior**:
No error

It should compile like the following code:
```CSharp
using System;

static ref readonly string CalcValue(out int x)
{
    x = 42;
    return ref Strings.s;
}

ref readonly var value =  ref CalcValue(out var x);

internal delegate ref readonly T0 ByRefFunc<T0>();

class Strings{
   public static readonly string s = "Something";
}
```

**Actual Behavior**:

> error CS8168: Cannot return local 'x' by reference because it is not a ref local
> error CS8347: Cannot use a result of 'CalcValue(out int)' in this context because it may expose variables referenced by parameter 'x' outside of their declaration scope

**Comment by bernd5** [comment]
btw.: readonly or not does not make any difference

**Comment by bernd5** [comment]
Sorry, I got the point.
